src/Polysemy/Internal/TH/Effect.hs

```diff
@@ -1,52 +1,56 @@
-{-# LANGUAGE OverloadedStrings #-}
-{-# LANGUAGE TemplateHaskell   #-}
-{-# LANGUAGE TupleSections     #-}
-
 {-# OPTIONS_HADDOCK not-home #-}
 
--- Originally ported from code written by Sandy Maguire (@isovector), available
--- at https://github.com/IxpertaSolutions/freer-effects/pull/28.
-
-{-|
-This module provides Template Haskell functions for automatically generating
-effect operation functions (that is, functions that use 'send') from a given
-effect algebra. For example, using the @FileSystem@ effect from the example in
-the module documentation for "Polysemy", we can write the following:
-
-@
-data FileSystem m a where
-  ReadFile :: 'FilePath' -> FileSystem 'String'
-  WriteFile :: 'FilePath' -> 'String' -> FileSystem ()
-'makeSem' ''FileSystem
-@
-
-This will automatically generate the following functions:
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE TemplateHaskell     #-}
+{-# LANGUAGE TupleSections       #-}
+{-# LANGUAGE MonadComprehensions #-}
```

Comment:
I think this is unused?

Reply:
Good point - why doesn't compiler warn about unused extensions with `-Wall`?

Reply:
Unsure.

`hlint` catches some of them.

Here's something I found on the subject: https://gitlab.haskell.org/ghc/ghc/issues/3085